### PR TITLE
Adopt `Sendable` for closures in `HTTPPipelineSetup.swift`

### PR DESCRIPTION
**Pull request by dnadoba**
and fix sendable warnings in tests.

### CODE REVIEW

**Review by Lukasa — CHANGES_REQUESTED** [review]

File: Sources/NIOHTTP1/HTTPPipelineSetup.swift [hunk 1]
```diff
@@ -77,7 +91,104 @@ extension ChannelPipeline {
...
+                                                                     leftOverBytesStrategy: leftOverBytesStrategy,
+                                                                     withClientUpgrade: upgrade)
+            }
+        }
+
+        return future
+    }
+    #endif
+    
+    private func _addHTTPClientHandlers(position: Position = .last,
```
[the remark this answers is not in the document]
Comment (dnadoba) [reply Sources/NIOHTTP1/HTTPPipelineSetup.swift:104]: Fixed in https://github.com/apple/swift-nio/pull/2214/commits/77c2c316d266ce4f1fb26e21d533b82097a865a3

File: Sources/NIOCore/NIOSendable.swift [hunk 2]
```diff
@@ -71,4 +58,6 @@ final class UnsafeMutableTransferBox<Wrapped> {
         self.wrappedValue = wrappedValue
     }
 }
+#if swift(>=5.5) && canImport(_Concurrency)
+extension UnsafeMutableTransferBox: @unchecked Sendable {}
```
Comment (dnadoba) [inline Sources/NIOCore/NIOSendable.swift:62]: The changes in `NIOSendable.swift` make `UnsafeTransfer` and `UnsafeMutableTransferBox` available in Swift 5.4. This allows us to unconditionally use them without code duplication and `#if`s.

File: Tests/NIOHTTP1Tests/HTTPServerUpgradeTests.swift [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 import XCTest
 import Dispatch
-import NIOCore
+@testable import NIOCore
```
Comment (dnadoba) [inline Tests/NIOHTTP1Tests/HTTPServerUpgradeTests.swift:17]: We could also copy & paste `UnsafeTransfer` and `UnsafeMutableTransferBox` into the test target if we don't want to add `@testable`.

**Review by Lukasa — CHANGES_REQUESTED** [review]

File: Tests/NIOHTTP1Tests/HTTPServerUpgradeTests.swift (hunk 3 above)
Comment (dnadoba) [reply Tests/NIOHTTP1Tests/HTTPServerUpgradeTests.swift]: Done in https://github.com/apple/swift-nio/pull/2214/commits/8472a2dd92301148daf017eb464ede8ec48f569e

**Review by Lukasa — APPROVED** [review]

**Pull request merged**